addons/fetchmail/models/fetchmail.py

```diff
@@ -190,10 +190,11 @@ def fetch_mail(self):
                         imap_server.store(num, '-FLAGS', '\\Seen')
                         try:
                             res_id = MailThread.with_context(**additionnal_context).message_process(server.object_id.model, data[0][1], save_original=server.original, strip_attachments=(not server.attach))
+                            imap_server.store(num, '+FLAGS', '\\Seen')
```

Comment:
Maybe use try-except-else for this, we do that a lot it helps isolate what we expect to fail from the rest.

Reply:
not sure i understand it clearly, should i move imap_server.store(num, '+FLAGS', '\\Seen') this line of code into the else condition after the except .
thanks @thiryr for the review and suggestion

Reply:
Hey
```
try:
     res_id = MailThread.with_context(**additionnal_context).message_process(server.object_id.model, data[0][1], save_original=server.original, strip_attachments=(not server.attach))
except Exception:
     imap_server.store(num, '-FLAGS', '\\Seen')
     ...
else:
     imap_server.store(num, '+FLAGS', '\\Seen')
```
In a try-except-else block, if `try` succeeds `else` will run. It's the same as putting everything in the try block but we explicitly specify which part we expect to throw an error.

In this case it's especially useful because we don't actually want to catch errors thrown by `server.store`. If we can't mark it as seen, we should not proceed (lest us process the same email twice).

Also feels like you would want to remove the `imap_server.store(num, '-FLAGS', '\\Seen')` you added. We already mark it 'unseen' at the start, and I think it's best to do that at the start.

-------------------------------------

fyi this one might be on hold for a couple of weeks as tde is on vacation, and it's the kind of stuff I think it's best for him to check out ^^

Reply:
hi @thiryr ,
i have updated as per the suggestion :)

regarding imap_server.store(num, '-FLAGS', '\\Seen') this line of code , i have removed this line of code and keep only one inside the exception condition.

on that day when i have error, the first one was not working i believe and thus i added the the same line of code inside the exception.

thanks

Reply:
Hm, I'm not sure. I assume there was a good reason to set it before message_process at the time at least. I would be inclined to just remove it because it should not be set in the first place if we search unseen.

bc0232cee63c2dbc79e2d9b93c8c10e034bc385f @tde-banana-odoo this is an oldie, any idea?

Reply:
Most likely, hence the original PR comment :p
`Only issue I could see is that if an email has an issue we're always going to re-process it.`

I think the big thing is that if it is a misconfiguration, you don't want to have to figure out which emails were marked as read and never processed. That's super annoying.

And if it's small, you just get a few problematic emails being reprocessed each time. For that maybe we can have some watchdog to keep the message ids of the last processed batch, and log a warning if it contains a mail that was already seen.

But it definitely complicates the process, sure.

Reply:
Then it is quite a different implementation ... we would like to do it (store failed incoming emails, in order to spot badly configured aliases for example). That would probably be something interesting to do for people using their own mail gateway with several aliases and lot of incoming emails... but otherwise you will end up fetching the same emails again and again, and this stack will grow up constantly probably.